.github/workflows/test.yml

```diff
@@ -0,0 +1,198 @@
+name: test
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
```

Comment:
any of these we'd want to split to say run a portion on prs to main only?

Reply:
I hope we don't have to go to partial CI, personally.  But I guess maybe.  Could be fed into the matrix generation if it made sense to handle it there.